### DISCUSSION TITLE
Add custom values

### DISCUSSION THREAD

**Discussion by asiag0** [mateusz-bak/openreads#709] [Ideas]
**Is your feature request related to a problem? Please describe.**
no

**Describe the solution you'd like**
adding custom values would add flexibility and extend the usability to inventory management.  for example, being able to add a custom value with a custom label, one could set the label to 'location', then enter a storage location into the value (such as 'grey book crate 2').  The custom field could be hidden automatically if it is not used, like is the case with other values (start/finish read, for example).

**Describe alternatives you've considered**
I'm currently using a Google Play app named "My Library" for inventory management.

**Additional context**
I am attempting to move away from google-managed software, so looking at F/OSS apps.  I have a number of self-hosted web apps, but want to keep my book inventory separate, since I would only need to access this data from a phone.  Openreads is amazing, but clearly the focus is towards tracking what you're reading/want to read.  I feel like the addition of a single custom value/field would allow me to use it for inventory management in a simple way, but which could provide flexibility to other users (like, 'times read', 'notes', or 'favorite quote').

**Comment by lucastucious** [comment]
I want to upvote this issue as it will fix many other issues like keeping track of owned books or adding publisher...

If we can assign theses value to open Library field it's even better !

**Comment by mateusz-bak** [comment]
I think we can add an extra parameter for a list of key - value pairs like JSON.

@lucastucious I think this would be a bit complicated but if you have any fields that are in Open Library and you would like to see in Openreads let me know.

**Comment by lucastucious** [comment]
wanted fields from OL :
- publisher
- series
- language

**Comment by JbIPS** [comment]
I made some tests with the OL API and the [input schema](https://github.com/internetarchive/openlibrary-client/blob/master/olclient/schemata/import.schema.json) states that the mandatory fields are :
- title
- source_records (don't know what to put here)
- authors
- publishers
- publish_date

So, if I look at the app now, only `publishers` is missing (I think `source_records` can be fiiled by the app, maybe that's what it meant).

Let me know where I can help to make this integration go further!
